server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -211,6 +226,17 @@ public TransportSearchAction(
         this.isRequestStatsEnabled = clusterService.getClusterSettings().get(SEARCH_REQUEST_STATS_ENABLED);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(SEARCH_REQUEST_STATS_ENABLED, this::setIsRequestStatsEnabled);
         this.searchRequestStats = searchRequestStats;
+        this.metricsRegistry = metricsRegistry;
+        this.searchQueryCategorizationEnabled = clusterService.getClusterSettings().get(SEARCH_QUERY_METRICS_ENABLED_SETTING);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(SEARCH_QUERY_METRICS_ENABLED_SETTING, this::setIsSearchQueryCategorizationEnabled);
+    }
+
+    private void setIsSearchQueryCategorizationEnabled(boolean isSearchQueryCategorizationEnabled) {
+        this.searchQueryCategorizationEnabled = isSearchQueryCategorizationEnabled;
+        if (this.searchQueryCategorizationEnabled && this.searchQueryCategorizer == null) {
```

Comment:
if ((this.searchQueryCategorizationEnabled  == true ) && this.searchQueryCategorizer == null) {